p2p/protocols/protocol.go

```diff
@@ -243,7 +243,7 @@ func (p *Peer) Run(handler func(ctx context.Context, msg interface{}) error) err
 // Drop disconnects a peer.
 // TODO: may need to implement protocol drop only? don't want to kick off the peer
 // if they are useful for other protocols
-func (p *Peer) Drop(err error) {
+func (p *Peer) Drop() {
```

Comment:
can we please preserve the error and log it or even put it in p2p.DiscSubprotocolError?
this is losing too much contextual info

Reply:
We could log the error, but I'd rather not propagate it down to p2p, and replace `DiscSubprotocolError` just now, as this is changing behaviour, and we already have too many changes.

The current change preserves the behavior we already have.

Reply:
@justelad I am not changing this behavior as well in this PR, let's keep it as small as possible.